### ISSUE TITLE
G.1.1(25.c/5): Inconsistent use of Nonblocking/Global for Pure library units

### ISSUE THREAD

**Issue by sttaft** [pat-rogers/Ada-202x-WG9-Informal-Review#48]
G.1.1(25.c/5) says:

> Ramification: {AI12-0241-1} A nongeneric pure unit is always nonblocking, so we don't need to talk about the Nonblocking aspect of these packages. Similarly, a nongeneric pure unit always has the Global aspect set to null, so we don't need to talk about that, either. 

The above is true, though there is no distinction between nongeneric and generic pure units at this point with respect to these rules (per 9.5(33/5)).  So one could rightly ask why we have continued to specify Nonblocking on various of the declared-Pure language-defined generic units.  Should we remove any remaining redundant declarations of Nonblocking (and Global=>null) from all declared pure language-defined units?  E.g. Unchecked_Conversion, Generic_[Constrained_]Array_Sort, Generic_Sort, Text_Buffers.Bounded, Generic_Elementary_Functions, Big_Numbers, Atomic_Operations*, Synchronized_Queue_Interfaces, Generic_Complex_Types, Generic_Complex_Elementary_Functions, Generic_Real_Arrays, Generic_Complex_Arrays

**Comment by ARG-Editor** [comment]
Of course, this changed relatively late; when those aspects were applied, there was no automatic application to generic units.

This note exists to explain why there isn't any explicit wording to specify Nonblocking and Global.

My personal preference would have been to specify Nonblocking and Global => null on all packages; I don't like the implicit rule much as it requires knowing about the special case for Pure. (And I think Pure itself is something that should fall out of use in new code, so it will be unusual to encounter it or care about it.) That's obviously too much work at this point, but removing them from many packages is also a heck of a lot of work.

I find this insufficiently broken to spend a lot of effort on at the last minute; it's not *wrong*. So the only change I made was to drop "nongeneric" from that Ramification. I put that into the AARM AI, AI12-0005-1.

If you really want to pursue changing that long list of packages, please ask to have the full ARG consider this question.
